From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Sun, 20 Mar 2016 12:56:59 +0100
Subject: [PATCH] {BUG new in 4.6-rc1} rtl8xxxu: initialize return value

clang:
drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu.c:5311:6: error:
      variable 'ret' is used uninitialized whenever 'if' condition is false
      [-Werror,-Wsometimes-uninitialized]
        if (!count) {
            ^~~~~~
drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu.c:5334:9: note:
      uninitialized use occurs here
        return ret;
               ^~~
drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu.c:5311:2: note:
      remove the 'if' if its condition is always true
        if (!count) {
        ^~~~~~~~~~~~
drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu.c:5284:16: note:
      initialize the variable 'ret' to silence this warning
        int count, ret;
                      ^
                       = 0
---
 drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu.c b/drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu.c
index cb6ce693dcce..4ac61b5c6238 100644
--- a/drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu.c
+++ b/drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu.c
@@ -5281,7 +5281,7 @@ static int rtl8723bu_active_to_emu(struct rtl8xxxu_priv *priv)
 	u8 val8;
 	u16 val16;
 	u32 val32;
-	int count, ret;
+	int count, ret = 0 /* BUG */;
 
 	/* Turn off RF */
 	rtl8xxxu_write8(priv, REG_RF_CTRL, 0);
-- 
